### ISSUE TITLE
`no-return-array-push`: reports "throwaway" short-circuit expressions

### ISSUE THREAD

**Issue by andreww2012** [sindresorhus/eslint-plugin-unicorn#3539]
I think this cannot be considered "using the push value":

```ts
declare const array: string[];
Math.random() > 0.5 && array.push(''); // Reported - it's not even assigned!
```

**Comment by sindresorhus** [comment]
I don't think we should make an exception for this. The return value is still used as the right-hand side of `&&`, even if the result of the full expression is discarded. This is just needlessly clever code; `if (condition) { array.push(''); }` is clearer. The rule is working as intended here.

**Comment by andreww2012** [comment]
Sorry, I don't fully understand how the return value is "used" in this example - it is not consumed in any way. I don't see this pattern as unsafe or dangerous.

On your style point - fair, I fully agree with it; however, these kind of expressions just exist in the codebase I'm working with and they all get flagged.

Your decision, of course, it just reads as a false positive to me.
